backend/domain/domainmodels.js

```diff
@@ -0,0 +1,119 @@
+class Room {
+    constructor() {
+
+        this.roomName = this.setRoomName(); // some function to initialise a roomID
+
+        // default values given.
+        this.gamestate = GameStateEnum["Day Time"];
+
+        this.maxPlayerCount = 6;
+
+        this.players = new Array(maxPlayerCount);
+
+        this.roundNumber = 0;
+
+        this.getRoomName = function () {
+            return this.roomName;
+        };
+
+        this.getGamestate = function () {
+            return this.gamestate;
+        };
+
+        this.setGamestate = function (gamestate) {
+            this.gamestate = gamestate;
+        };
+
+        this.getPlayers = function () {
+            return this.players;
+        };
+
+        this.addPlayer = function (player) {
+            if (player !== null &&
+                this.getPlayers.length <= maxPlayerCount) {
```

Comment:
`getPlayers().length` . Dont need this for methods

Reply:
fixed.